### ISSUE TITLE
Mixed Precision training is ~10 times slower

### ISSUE THREAD

**Issue by Hazarapet** [tensorflow/tensorflow#41505]
Mixed Precision training is too slow when I use `mixed_float16` policy
**System information**
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow):**
- **Ubuntu 18.04:**
- **TensorFlow installed from pip3:**
- **TensorFlow version 2.2.0:**
- **Python version 3.7:**
- **CUDA version 10.1 cuDNN version 7.5:**
- **GPU model and memory RTX 2080ti 11gb:**

I've designed a segmentation model, which consists of different regular layers like BatchNorm, Conv2D, Activation, etc.. Haven't designed and used any custom layer.
I'm using binary cross-entropy as a loss function and also put `float32` dtype on my network's outputs (like to official doc says), which is a sigmoid function. Data loading is with `float32` as well but **when I train with mixed precision, it gets super slow ~6-10 times**. I use the same network with the same batch size, without changing anything else. I've calculated the time between the processes and here is what it looks like

- **Feedforward is slower ~10x**
- **Loss Computing is slower ~6x**
- **Gradients computing is slower ~6x**

I assume it should be faster than the `float32` policy, but it turns out not. It's super slow. At least I'm expecting it should not be slower and the main advantage would be to take a bigger **batch size**.

**Comment by Hazarapet** [comment]
I think I've found the problem. **It's because of the resize operation I've got**.
It turns out **tf.image.resize** or **keras.Upsample2D** (or any sampling approach, which uses image resize from keras backend), breaks the `mixed-precision` training regime.

You can use **Conv2D** or **Pooling2D** for **downsampling** and **Con2DTranspose** for upsampling to avoid this problem.

What about this issue? In some situations, we have to use a resize operation. How to make it work as well?

**Comment by lordtt13** [comment]
Have you tried using keras.layers.Reshape?

**Comment by Hazarapet** [comment]
No, I haven't tried. 
But how the to reshape is going to fill the missed values? cause interpolation aims to upsample (bigger resolution) and fill additional sampling values.

**Comment by lordtt13** [comment]
If you are interpolating then reshape is useless and as you suggested to gain best performance Conv2DTranspose will be the best way to go when upsampling. A doubt though, why would you explicitly require tf.image.resize or keras.Upsample2D if you can achieve the same with better performance using Conv2DTranspose?

**Comment by lordtt13** [comment]
Also are you sure that tf.image.resize supports mixed policy?

**Comment by Hazarapet** [comment]
I couldn't find a note which would tell that `tf.image.resize` does not support `mixed-precision`. Also, **Conv2DTransponse** is not always a good choice. I tried both, but Conv2DTranspose is really worst than the interpolation (in my task).

**Comment by lordtt13** [comment]
Then van you use tf.image.resize and Conv2DTranspose in ciinjunction? i.e. Using Conv2DTranspose as part of the model and tf.image.resize outside of it.

**Comment by Hazarapet** [comment]
I don't think I can use it. it's inside my decoder and each layer has an upsampling op.

**Comment by lordtt13** [comment]
Ah shoot, I'm stumped.

**Comment by Hazarapet** [comment]
Here is an issue which was opened a couple of years ago "**Provide GPU implementations for resize image methods**" #533
Closed without any solution.

**Comment by amahendrakar** [comment]
@Hazarapet,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here. Thanks!

**Comment by Hazarapet** [comment]
@amahendrakar thank you for your response.
I'm sorry but I can't share the code. It's a model, containing regular layers like **batchnorm**, **conv2d**, **activation**, and **sampling** layers and **nothing else**. I don't have any complicated custom implementations in my model. The main problem is related to sampling operations. I'm guessing image resize does not have GPU support for `mixed_float16`. If I replace them with Conv2d and Conv2dTranspose, the training process gets really fast.

**Comment by amahendrakar** [comment]
@Hazarapet,
Without a sample code it is difficult for us to pinpoint the issue. Is it possible for you to provide a mock code which replicates the behavior? So that it'll will allow us to determine the source of the issue. Thanks!

**Comment by reedwm** [comment]
Note this may have been fixed by https://github.com/tensorflow/tensorflow/commit/67d15573a776119d5a544ed266dc2514ae13c3b5.
